notebooks/ndd_example.ipynb

```diff
@@ -0,0 +1,223 @@
+    "\n",
+    "- All 1911 PDFs (~4 hours): `docker compose run sycamore_crawler_s3 aryn-public cccmad`\n",
+    "- 35 PDFs needed for demo (minutes): `docker compose run sycamore_crawler_s3 aryn-public cccmad-tiny`\n",
+    "\n",
+    "Note that for the full dataset, approximately 17 documents will fail to ingest for various reasons.  The Sycamore importer will keep retrying them, but the problems will persist.  It's not necessary for 100% of the documents to be ingested in order to run this example.  Once the importer queue shrinks to ~100 documents, it's OK to proceed.  Note that due to variations in ingestion, results may not exactly match what's described here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "bbf93d45-bee3-4202-8d5a-88497196db06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import requests\n",
+    "import warnings\n",
+    "import urllib3\n",
+    "warnings.filterwarnings(\"ignore\", category=urllib3.exceptions.InsecureRequestWarning)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a9e0eb6c-2585-4c00-97ea-f7f7aa316ed5",
+   "metadata": {},
+   "source": [
+    "The code below exists to retrieve the embedding model ID from OpenSearch.  This ID is different every time OpenSearch is set up.  We need to supply the ID in our query.  So, we need to fetch it every time in order to be sure."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "06e9a476-134f-4901-a89e-3c91bdb8881c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_model_id():\n",
+    "    query = {\n",
+    "        'query': {\n",
+    "            'bool': {\n",
+    "                'must': [\n",
+    "                    {\n",
+    "                        'match': {'name': 'all-MiniLM-L6-v2'},\n",
+    "                    },\n",
+    "                    {\n",
+    "                        'term': {'model_config.model_type': 'bert'},\n",
+    "                    },\n",
+    "                ],\n",
+    "            },\n",
+    "        },\n",
+    "    }\n",
+    "    with requests.get('https://opensearch:9200/_plugins/_ml/models/_search', json=query, verify=False) as resp:\n",
+    "        res = json.loads(resp.text)\n",
+    "        return res['hits']['hits'][0]['_id']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c94d8d48-a8b6-4704-a709-5cd022ed7343",
+   "metadata": {},
+   "source": [
+    "This next function performs the supplied query and prints out both the retrieved chunks and the AI-generated answer.  For clarity, the text chunks are truncated at 80 characters."
```

Comment:
`<br><hr>`

at the beginning of each of the markdown sections.

Reply:
It turns out <br><hr> messes up the backtick formatting.  <br> and --- works better.